src/vivarium_cluster_tools/psimulate/worker/vivarium_work_horse.py

```diff
@@ -93,9 +93,7 @@ def work_horse(job_parameters: dict) -> Tuple[pd.DataFrame, Dict[str, pd.DataFra
                 f'Simulant initialization completed in {exec_time["simulant_initialization_minutes"]:.3f} minutes.'
             )
             start_time = pd.Timestamp(**sim.configuration.time.start.to_dict())
-        end_time = pd.Timestamp(**sim.configuration.time.end.to_dict())
-        step_size = pd.Timedelta(days=sim.configuration.time.step_size)
-        num_steps = int(math.ceil((end_time - start_time) / step_size))
+        num_steps = sim.get_number_of_steps(start_time)
```

Comment:
We shouldn't be providing start time to this method.

Reply:
But we have to get it from the Parallel sim context or the backups

Reply:
On the line above, it gets the start time from `sim.configuration.time.start` Isn't that just the model spec start time? Or does this value actually change in the event you're starting from a backup?